code/modules/economy/worth_cash.dm

```diff
@@ -164,36 +164,147 @@
-	icon_state = "efundcard"
-	desc = "A card that holds an amount of money."
-
+/obj/item/charge_stick
+	name = "charge-stick"
+	icon = 'icons/obj/items/credstick.dmi'
+	on_mob_icon = 'icons/obj/items/credstick.dmi'
+	icon_state = "peasant"
+	desc = "A digital stick that holds an amount of money."
+
+	var/max_worth = 5000
 	var/loaded_worth = 0
-	var/owner_name = "" //So the ATM can set it so the EFTPOS can put a valid name on transactions.
+	var/creator			// Who originally created this card. Mostly for book-keeping purposes. In game these cards are 'anonymous'.
+	var/id = "" 		//So the ATM can set it so the EFTPOS can put a valid name on transactions.
 	var/currency
+	var/lock_type = /datum/extension/lockable/charge_stick
+	var/grade = "peasant"
 
-/obj/item/charge_card/Initialize(ml, material_key)
+/obj/item/charge_stick/Initialize(ml, material_key)
 	. = ..()
+	id = "[grade]-card-[sequential_id("charge_stick")]"
 	appearance_flags |= PIXEL_SCALE
 	if(!ispath(currency, /decl/currency))
 		currency = GLOB.using_map.default_currency
+		update_name_desc()
+	set_extension(src, lock_type)
+	update_icon()
+
+/obj/item/charge_stick/proc/update_name_desc()
+	if(!isnull(currency))
+		var/decl/currency/cur = decls_repository.get_decl(currency)
+		name = "[cur.name]-stick"
+		desc = "A pre-charged digital stick that anonymously holds an amount of [cur.name]."
 
-/obj/item/charge_card/proc/adjust_worth(amt)
+/obj/item/charge_stick/proc/adjust_worth(amt)
 	loaded_worth += amt
 	if(loaded_worth < 0)
 		loaded_worth = 0
 
-/obj/item/charge_card/examine(mob/user, distance)
+/obj/item/charge_stick/examine(mob/user, distance)
 	. = ..(user)
 	if(distance <= 2 || user == loc)
-		to_chat(user, SPAN_NOTICE("<b>Owner:</b> [owner_name]."))
-		var/decl/currency/cur = decls_repository.get_decl(currency)
-		to_chat(user, SPAN_NOTICE("<b>[capitalize(cur.name)]</b> remaining: [Floor(loaded_worth / cur.absolute_value)]."))
+		var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+		if(lock.locked)
+			to_chat(user, SPAN_WARNING("\The [src] is locked."))
+		else
+			to_chat(user, SPAN_NOTICE("<b>Id:</b> [id]."))
+			var/decl/currency/cur = decls_repository.get_decl(currency)
+			to_chat(user, SPAN_NOTICE("<b>[capitalize(cur.name)]</b> remaining: [Floor(loaded_worth / cur.absolute_value)]."))
 
-/obj/item/charge_card/get_base_value()
+/obj/item/charge_stick/get_base_value()
 	. = holographic ? 0 : loaded_worth
 
+/obj/item/charge_stick/attackby(var/obj/item/W, var/mob/user)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+
+	if(istype(W, /obj/item/charge_stick))
+		var/obj/item/charge_stick/sender = W
+		var/datum/extension/lockable/W_lock = get_extension(W, /datum/extension/lockable)
+		if(lock.locked)
+			to_chat(user, SPAN_NOTICE("Cannot transfer funds to a locked [src]."))
+			return TRUE
+		if(W_lock.locked)
+			to_chat(user, SPAN_NOTICE("Cannot transfer funds from a locked [W]."))
```

Comment:
This is purely subjective but I always found messages like this to be kind of distracting - the sort of anonymous system messages being piped directly into the chat. I'd suggesting putting `\icon[src]` at the start or making it '\The [src] beeps, "Cannot foo bar."`.